src/components/page/FoodsPage.tsx

```diff
@@ -1,3 +1,156 @@
+    setCategoryList(categoryList);
+    setQuery(searchParams?.get('query') || '');
+
+    const filterByGroup = (row: { groupId: string }) => group === '' || group === row.groupId;
+    const filterByContainer = (row: { id: string }) => container === '' || container === row.id;
+    const filterByCategory = (food: { category: string }) => {
+      if (!categoryList.length) return true;
+      return categoryList.some((c) => food.category.includes(c));
+    };
+    const filterByName = (food: { name: string }) =>
+      food.name.toLowerCase().includes(query.toLowerCase());
+
+    const filteredContainers = containers.filter(filterByGroup).filter(filterByContainer);
+    const initialFoodsView: IFoodView[] = filteredContainers.flatMap((container: IContainer) =>
+      container.foods.map((food: IFood) => ({
+        ...food,
+        container: container.name,
+      })),
+    );
+    const filteredFoods = initialFoodsView.filter(filterByName).filter(filterByCategory);
+
+    const sortFoods = (a: IFood, b: IFood) => {
+      switch (sort) {
+        case 'name':
+          return a.name.localeCompare(b.name);
+        case 'expiry':
+          return a.expiry < b.expiry ? -1 : 1;
+        default:
+          return a.createdAt > b.createdAt ? -1 : 1; // default is createdAt desc order
+      }
+    };
+
+    const sortedFoods = [...filteredFoods].sort(sortFoods);
+    setDisplayedFoods(sortedFoods);
+  }, [containers, query, sort, group, container, searchParams]);
+
+  return (
+    <div className="mt-6 mx-4 mb-16">
+      <div className="relative">
+        <SearchBar />
+        <FilterButton
+          containers={containersGroupByGroups}
+          containerIdMap={containerIdMap}
+          groupIdMap={groupIdMap}
+        />
+      </div>
+      <div
+        className={`flex ${
+          group || container || categoryList.length > 0
+            ? 'gap-1.5 mt-4 mb-1.5 sm:overflow-auto sm:whitespace-normal overflow-scroll whitespace-nowrap'
+            : ''
+        }`}
+      >
+        {group ? (
+          <FilterBadge
+            icon={HomeIcon_Off}
+            text={groupIdMap[group]}
+            onCrossClick={() => removeGroupFilter()}
+          />
+        ) : (
+          <></>
+        )}
+        {container ? (
+          <FilterBadge
+            icon={ContainerIcon}
+            text={containerIdMap[container]}
+            onCrossClick={() => removeContainerFilter()}
+          />
+        ) : (
+          <></>
+        )}
+        {categoryList.map((key) => {
+          return (
+            <FilterBadge
+              key={key}
+              emoji={foodCategories[key]?.emoji}
+              text={foodCategories[key]?.name}
+              onCrossClick={() => removeCategoryFilter(key)}
+            />
+          );
+        })}
+      </div>
+      <div className="flex items-center justify-end">
+        <FoodSort />
+        <div className="px-3.5">
+          <Icon icon={MenuMeatballIcon} size={4} />
+        </div>
```

Comment:
This meatball button should take up 48x48 px:

<img width="250" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/95d09eb1-8812-4b02-a45c-882685a497e5">

<img width="250" alt="image" src="https://github.com/genesis-tech-tribe/nishiki-frontend/assets/99148565/9ddd73fc-6ea7-423d-bd23-ccd685ce777c">